tests/test_browser.py

```diff
@@ -2713,12 +2713,18 @@ def test_emterpreter_async_iostream(self):
     self.btest('emterpreter_async_iostream.cpp', '1', args=['-s', 'EMTERPRETIFY=1', '-s', 'EMTERPRETIFY_ASYNC=1'])
 
   def test_modularize(self):
-    for opts in [[], ['-O1'], ['-O2', '-profiling'], ['-O2']]:
+    for opts in [[], ['-O1'], ['-O2', '-profiling'], ['-O2'], ['-O2', '--closure', '1']]:
       for args, code in [
         ([], 'Module();'), # defaults
         (['-s', 'EXPORT_NAME="HelloWorld"'], '''
           if (typeof Module !== "undefined") throw "what?!"; // do not pollute the global scope, we are modularized!
           HelloWorld();
+          if (HelloWorld.Pointer_stringify) { // module constructor should not be polluted
+            var xhr = new XMLHttpRequest();
```

Comment:
Hmm, yes, I see, this tests after `HelloWorld()` is called. Good point. But it seems like if HelloWorld already sent a response, then your xhr is also too late? Or at best races with it?

It seems like the safe thing might be to add a method in the C file, "report_success" or such. It would be called manually on success, so an error would lead to it not being called. I hope that's simple, but if that seems like a complicated change to do in this PR, then we can leave it for later.